app/controllers/parsers_controller.rb

```diff
@@ -85,9 +93,21 @@ def allow_flush
     end
   end
 
+  private
+
   def parser_params
     params
       .require(:parser)
       .permit(:name, :partner, :source_id, :strategy, :parser_template_name, :message, :content)
   end
+
+  def datatable_params
+    {
+      search:    params[:search][:value],
+      start:     params[:start].to_i,
+      per_page:  params[:length].to_i,
+      order_by:  params[:columns][params[:order]['0'][:column]][:data],
```

Comment:
this line look crazy hey Paul 😬

Reply:
Yeah I know, it's because of the datatable params. I'll try to make it more understandable.

Reply:
Is it a bit clearer?

Reply:
👍